sources/packages/backend/apps/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -171,22 +176,44 @@ export class ApplicationStudentsController extends BaseController {
     @Param("applicationId", ParseIntPipe) applicationId: number,
     @UserToken() studentToken: StudentUserToken,
   ): Promise<void> {
+    const isFulltimeAllowed = this.configService.isFulltimeAllowed;
     const programYear = await this.programYearService.getActiveProgramYear(
       payload.programYearId,
     );
+    if (
+      ![OfferingIntensity.fullTime, OfferingIntensity.partTime].includes(
+        payload.data.howWillYouBeAttendingTheProgram,
+      )
+    ) {
+      throw new UnprocessableEntityException(
+        "Offering intensity type is invalid.",
+      );
+    }
     if (!programYear) {
       throw new UnprocessableEntityException(
         "Program Year is not active. Not able to create an application invalid request.",
       );
     }
-
+    if (
+      !isFulltimeAllowed &&
+      payload.data.howWillYouBeAttendingTheProgram ===
```

Comment:
Please keep in mind that while dealing with `payload.data` it can be cast to the `ApplicationData` to avoid the `any` along this method.